lib/algorithms/rsa.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const crypto = require('crypto');
+const { promisify } = require('util');
+
+const algorithms = require('../algorithms');
+const {
+  InvalidAccessError,
+  NotSupportedError
+} = require('../errors');
+const { kKeyMaterial, CryptoKey } = require('../key');
+const { limitUsages, opensslHashFunctionName, toBuffer } = require('../util');
+
+const generateKeyPair = promisify(crypto.generateKeyPair);
+
+function uint8ArrayToUint32(bigInteger) {
```

Comment:
is this a normal js array with numbers as values, or a Uint8Array? If the latter, cannot readUInt32BE() work?

There are no checks for overflow, if `bigInteger` has `size > 4`, then it will overflow 32 bits. Is that not a problem?

Reply:
It is an `Uint8Array`, but its length can be shorter than four bytes, and it can also contain leading zeros. I believe `readUInt32BE` would throw for three-byte arrays.

Yes, we should add some kind of input validation. (I skipped that part for most APIs so far.)

Reply:
I'm just skimming, so I might have missed the input validation skipping that you are doing. I  thought you were pretty consistently throwing on unexpected types, for example.

Anyhow, I'd at least add an `assert()` here on the input array length being < 5.

And I confirmed, yes, Buffer requires exactly a length of 4. Maybe that could be arranted, or is the input here from the user?

Reply:
I will add a check. To be honest, there are not even unit tests for the error paths, so there is still a lot of work to be done regarding input validation and error handling.

The input is from the user, it's the `publicExponent: new Uint8Array([...])` option. The WebCrypto API was designed without `BigInt`, so they decided to use `Uint8Array` for big integers.